packages/admin-frontend/pages/crop.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+
+import React, { Component } from 'react';
+import Cropper from 'react-cropper';
+import 'cropperjs/dist/cropper.css';
+import fetch from 'isomorphic-fetch';
+import { imageApiUrl } from '../config';
+import { getTokenFromLocalCookie } from '../lib/fetch';
+
+type Props = {
+  imageUrl?: string
+};
+
+type State = {
+  ratio: number,
+  imageApiBody?: any,
+  existingParameters?: any,
+  postResult?: string
+};
+
+class Crop extends Component<Props, State> {
+  state = { ratio: 0.81, imageApiBody: null, existingParameters: null };
+
+  toPercentages = c => {
+    var cropStartPxX = c.getCropBoxData().left - c.getCanvasData().left;
+    var cropEndPxX = cropStartPxX + c.getCropBoxData().width;
+    var cropStartPxY = c.getCropBoxData().top - c.getCanvasData().top;
+    var cropEndPxY = cropStartPxY + c.getCropBoxData().height;
+
+    return {
+      cropStartX: Math.max(
+        0,
+        Math.round((cropStartPxX / c.getImageData().naturalWidth) * 100)
+      ),
+      cropEndX: Math.min(
+        100,
+        Math.round((cropEndPxX / c.getImageData().naturalWidth) * 100)
+      ),
+      cropStartY: Math.max(
+        0,
+        Math.round((cropStartPxY / c.getImageData().naturalHeight) * 100)
+      ),
+
+      cropEndY: Math.min(
+        100,
+        Math.round((cropEndPxY / c.getImageData().naturalHeight) * 100)
+      )
+    };
+  };
+
+  toImageApiBody = pcnt => {
+    const existingParametersForCurrentRatio =
+      this.state.existingParameters &&
+      this.state.existingParameters.find(
+        p => p.forRatio === String(this.state.ratio)
+      );
+    const revision =
+      (existingParametersForCurrentRatio &&
+        existingParametersForCurrentRatio.revision) ||
+      1;
+
+    return {
+      forRatio: String(this.state.ratio),
+      revision: revision,
+      imageUrl:
+        this.props.imageUrl &&
+        this.props.imageUrl.substr(this.props.imageUrl.lastIndexOf('/')),
+      rawImageQueryParameters: {
+        cropStartX: pcnt.cropStartX,
+        cropEndX: pcnt.cropEndX,
+        cropStartY: pcnt.cropStartY,
+        cropEndY: pcnt.cropEndY
+      }
+    };
+  };
+
+  _crop() {
```

Comment:
Curious as to why this method is "marked private" as opposed to the other methods

Reply:
No reason. Fixed.